test/spec/AuthStateManager.js

```diff
@@ -57,46 +59,54 @@ describe('AuthStateManager', () => {
     };
   });
 
+  afterEach(() => {
+    jest.useRealTimers();
+  });
+
   describe('constructor', () => {
     it('should listen on "added" and "removed" events from TokenManager', () => {
       new AuthStateManager(sdkMock); // eslint-disable-line no-new
       expect(sdkMock.tokenManager.on).toHaveBeenCalledWith('added', expect.any(Function));
       expect(sdkMock.tokenManager.on).toHaveBeenCalledWith('removed', expect.any(Function));
     });
 
-    it('should call updateAuthState when "added" event emitted', () => {
+    it('should call updateAuthState when "added" event emitted', (done) => {
       const instance = new AuthStateManager(sdkMock);
-      instance.updateAuthState = jest.fn();
+      jest.spyOn(instance, 'updateAuthState');
       instance._setLogOptions = jest.fn();
       sdkMock.emitter.emit('added', 'fakeKey', 'fakeToken', { timestamp: 111 });
-      expect(instance._setLogOptions).toHaveBeenCalledWith({ event: 'added', key: 'fakeKey', token: 'fakeToken' });
-      expect(instance.updateAuthState).toHaveBeenCalled();
+      instance.subscribe(() => {
```

Comment:
instead of using subscribe/done, can we use jest fake timers and advance all timers so that this can run synchronosly?

Reply:
tried, looks like the debounce does not like the `jest.runAllTimers()`... it causes infinite loop..